java/fury-core/src/main/java/org/apache/fury/serializer/ArraySerializers.java

```diff
@@ -69,12 +76,23 @@ public ObjectArraySerializer(Fury fury, Class<T[]> cls) {
       if (fury.getClassResolver().isMonomorphic(componentType)) {
         if (fury.isCrossLanguage()) {
           this.componentTypeSerializer = null;
+          this.collectionSerializer = null;
+          this.collectionGenericType = null;
         } else {
           this.componentTypeSerializer = fury.getClassResolver().getSerializer(componentType);
+          if (dimension > 1) {
+            this.collectionSerializer = new FuryArrayAsListSerializer(fury);
+            this.collectionGenericType = buildCollectionGenericType(dimension);
+          } else {
+            this.collectionSerializer = null;
```

Comment:
Why it's null for 1d object array?

Reply:
I thought for 1D / Non nested generics that we intended to use the normal default serialization path that was in place before?

Reply:
That is only for primitive array. Other array still use Collection serialization protocol. `List<String>` and `String[]` should have same layout

Reply:
Ok I  am under impression T[] can never have primitives only wrapper primitive classes should those use predvious method as well ie. Double[] ?